### PR TITLE
Basic profile editing feature added with new model Profile.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
User is now able to view more information of his/her profile than before. User is also now able to edit any profile information about himself/herself as he/she wishes.

### CODE REVIEW

**Comment by saadyousafarbi** [comment]
@zubair-arbi Please review changes.

File: autoshop/core/models.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    bio = models.TextField(
+        help_text='Some details about the user.', max_length=500, null=True
+    )
+    gender = models.CharField(
+        max_length=10, help_text='Specify gender.', choices=GENDER_CHOICES, null=True
+    )
+    photo = models.ImageField(help_text='Profile picture.', null=True)
+    date_of_birth = models.DateField(help_text='Date of birth.', null=True)
+    mobile_number = models.CharField(
+        help_text='Verified mobile number with country code', max_length=12, null=True
```
Comment (zubair-arbi) [inline autoshop/core/models.py]: lets set it to 15 length for now

File: autoshop/core/models.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    )
+    address = models.CharField(help_text='Local address of user.', max_length=250, null=True)
+    city = models.CharField(max_length=100, null=True)
+    country = models.CharField(max_length=100, null=True)
+
+    objects = models.Manager()
```
Comment (zubair-arbi) [inline autoshop/core/models.py]: please remove this, we dont need it

File: autoshop/core/templates/editprofile.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<head>
+    <meta charset="UTF-8">
+    <title>Autoshop | Profile</title>
+</head>
+<div class="jumbotron" style="box-shadow: 2px 2px 4px #000000;">
+        <h2 class="text-center"> Auto-Shop</h2>
+        <h3 class="text-center">A website with information about automobiles for buying/selling purposes.</h3>
+</div>
+            <form method="POST">
+                            {% csrf_token %}
```
Comment (zubair-arbi) [inline autoshop/core/templates/editprofile.html]: please make sure we have indented all files properly

File: autoshop/core/templates/profile.html [hunk 4]
```diff
@@ -7,8 +7,14 @@ <h2 class="text-center"> Auto-Shop</h2>
         <h3 class="text-center">A website with information about automobiles for buying/selling purposes.</h3>
 </div>
     Username: {{user_info_dict.username}} <br>
-    First Name: {{user_info_dict.first_name}} <br>
-    Last Name:  {{user_info_dict.last_name}} <br>
-    Email: {{user_info_dict.email}} <br>
+    Bio: {{user_info_dict.bio}} <br>
```
Comment (zubair-arbi) [inline autoshop/core/templates/profile.html]: `{{user_info_dict.bio}}` -> `{{ user_info_dict.bio }}` notice spaces inside curly brackets

File: autoshop/core/templates/profile.html [hunk 5]
```diff
@@ -7,8 +7,14 @@ <h2 class="text-center"> Auto-Shop</h2>
...
+    Gender:  {{user_info_dict.gender}} <br>
+    Mobile number: {{user_info_dict.mobile_number}} <br>
+    Date of birth: {{user_info_dict.date_of_birth}} <br>
+    Address: {{user_info_dict.address}} <br>
+    City: {{user_info_dict.city}} <br>
+    Country: {{user_info_dict.country}} <br>
+
+    <a href="/editprofile">Edit profile</a>
```
Comment (zubair-arbi) [inline autoshop/core/templates/profile.html]: use django's reverse url, avoid using hard coded values

File: autoshop/core/urls.py [hunk 6]
```diff
@@ -11,4 +11,5 @@
     url(r'^login', views.log_in, name='login'),
     url(r'^logout', views.log_out, name='log_out'),
     url(r'^profile', views.profile, name='profile'),
+    url(r'^editprofile', views.edit_profile, name='edit_profile'),
```
Comment (zubair-arbi) [inline autoshop/core/urls.py]: this url should be `/profile/edit`

File: autoshop/core/views.py [hunk 7]
```diff
@@ -1,11 +1,11 @@
 from __future__ import unicode_literals
 
-from django.contrib.auth import authenticate, login
+from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth.models import User
-from django.contrib.auth import logout
 from django.shortcuts import render, redirect
-from .forms import SignupForm, SigninForm
+from .forms import SignupForm, SigninForm, EditProfileForm
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: use absolute urls

File: autoshop/core/views.py [hunk 8]
```diff
@@ -95,12 +95,53 @@ def profile(request):
...
+        if form.is_valid():
+            Profile.objects.filter(user=username).update(
+                bio = form.data['bio'],
+                gender = form.data['gender'],
+                mobile_number = form.data['mobile_number'],
+                address = form.data['address'],
+                city = form.data['city'],
+                country = form.data['country'],
+            )
+            return render(request, 'editprofile.html', context)
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: after updating profile, the user should be redirected to profile view page.

File: autoshop/core/views.py [hunk 9]
```diff
@@ -95,12 +95,53 @@ def profile(request):
...
+        'country': user_information.country,
     }
     context = {'user_info_dict': user_info_dict}
     return render(request, 'profile.html',  context)
+
+
+@login_required()
+def edit_profile(request):
+    """
+    This view directs user to edit page where user can change profile
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: update docstring as this will not always redirect user to profile edit page

File: autoshop/core/forms.py [hunk 10]
```diff
@@ -31,3 +33,19 @@ class SigninForm(forms.Form):
     username = forms.CharField(max_length=100)
     password = forms.CharField(widget=forms.PasswordInput, required=True)
 
+
+class EditProfileForm(ModelForm):
+    """
+    Edit Profile form.
+
+    fields:
+        all fields from Profile model
```
Comment (zubair-arbi) [inline autoshop/core/forms.py]: remove it as it is not providing any useful information.  we can clearly see that its a `Profile` model form

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
I have gone through changes and added my initial feedback.

**Review by zubair-arbi — APPROVED** [review]
Changes look good now 👍

**Pull request merged**